### ISSUE TITLE
Require sparse gradient clipping

### ISSUE THREAD

**Issue by gavin1332** [PaddlePaddle/Paddle#9048]
Sparse gradient clipping is helpful in many NLP sequential tasks, such as machine translation, language model, semantic role labeling, etc., which is applied on embedding matrix to shrink the gradient accumulated along the sequence to avoid gradient exploding.
We found a gradient clipping operator for dense matrix, but it is not suitable for sparse-updated embedding, so we need the sparse version.

Besides, L1/L2 regularization and Ada- series optimization algorithm should be also adapted for sparse updating.

**Comment by jacquesqiao** [comment]
```python
emb = fluid.layers.embedding(input=src_wordseq,
        size=[vocab_size, emb_size],
        param_attr=fluid.ParamAttr(
            initializer=fluid.initializer.Uniform(low=-0.05, high=0.05),
            learning_rate=emb_lr_x,
            gradient_clip=fluid.clip.ClipByValue(10, 0.1)),
        is_sparse=True)
```
ClipByValue is supported now.

will check ClipByNorm and ClipByGlobalNorm.

**Comment by JiayiFeng** [comment]
Our gradient clip use `clip_op` and `clip_by_norm_op`. They don't support sparse matrix so far. So gradient clip cannot apply to sparse parameters now.
